src/Workspaces/Core/Portable/Workspace/Solution/Document.cs

```diff
@@ -453,7 +453,8 @@ internal virtual Document WithFrozenPartialSemantics(CancellationToken cancellat
             if (solution.PartialSemanticsEnabled &&
                 this.Project.SupportsCompilation)
             {
-                var newSolution = this.Project.Solution.WithFrozenPartialCompilationIncludingSpecificDocument(this.Id, cancellationToken);
+                var linkedDocumentIds = this.GetLinkedDocumentIds();
+                var newSolution = this.Project.Solution.WithFrozenPartialCompilationIncludingSpecificDocument(this.Id, linkedDocumentIds, cancellationToken);
```

Comment:
Since we're already passing in the DocumentId for the primary document, can we just fetch the linked DocumentIds further down? I'm not sure why we're fetching it here only to pass it through.

Reply:
Couldn't see a way to get it from a ```DocumentState```

Reply:
The Solution level method is https://github.com/dotnet/roslyn/blob/475293444be097e8a7cab8658b99d3c706b6e8e6/src/Workspaces/Core/Portable/Workspace/Solution/Solution.cs#L1697 although that'd be easy to move to SolutionState. It looks like that one also does keep the original ID in the list, so that'll make the rest of the code easier too.